app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
+  def warnings
+    [].tap do |warnings|
+      warnings << Notice.new(problem_description_key, :warning) if problem?
```

Comment:
Couldn't this just be

```ruby
[Notice.new(problem_description_key, :warning)] if problem?
```

There's only ever going to be one `Notice` object in the array, right?

I think the `tap` weirds me out here and above because my brain subconsciously expects us to be iterating over something, because the pattern looks too much like an `each` or a `map` kind of operation.